### PR TITLE
Disable Automated Time Control

### LINKED ISSUE #687: Disable Automated Time Control
# For New Contributors
[How to contribute](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/wiki/Getting-Started-as-a-Contributor)

## Description

Currently there are quite a few ways time is controlled automatically

Here are some of those cases
- When on fast forward and clicking on the map to move your party
- Entering settlements
- etc

Disable all time control except for clicking the buttons on the bottom of the screen and 1, 2, 3 keyboard controls

## Intended Design

N/A

## Location

Create a new branch based from development

## Related Issues

N/A

## Requirements

N/A

## Additional information

N/A

## Definition of Done
- [ ] Class level comments exist for all new classes.
- [ ] XUnit tests exist for every method that does not require the game to be ran.

### PR DESCRIPTION
**Pull request by EgardA**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] All new classes have class-level documentation comments, if there are any at all
- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:

## What is the new behaviour?

Resolves #687 

## Other information

### CODE REVIEW

File: source/GameInterface/Services/Time/Handlers/TimeControlHandler.cs [hunk 1]
```diff
@@ -29,6 +29,10 @@ private void Handle(MessagePayload<SetTimeControlMode> obj)
     {
         var payload = obj.What;
         CampaignTimeControlMode newTimeMode = (CampaignTimeControlMode)payload.NewTimeMode;
+
+        if (newTimeMode == CampaignTimeControlMode.StoppablePlay) { newTimeMode = CampaignTimeControlMode.UnstoppablePlay; }
+        if (newTimeMode == CampaignTimeControlMode.StoppableFastForward) { newTimeMode = CampaignTimeControlMode.UnstoppableFastForward; }
+
```
Comment (garrettluskey) [inline source/GameInterface/Services/Time/Handlers/TimeControlHandler.cs:35]: I don't understand how these changes solve the listed issue. The seems to only convert our interface time controls with the games

**Review by garrettluskey — CHANGES_REQUESTED** [review]

File: source/GameInterface/Services/Time/Handlers/TimeControlHandler.cs (hunk 1 above)
Comment (EgardA) [reply source/GameInterface/Services/Time/Handlers/TimeControlHandler.cs:35]: Every time Change in game does a isNot check for unstoppable time. If we always set it as unstoppable we need less patches.

Comment (EgardA) [reply source/GameInterface/Services/Time/Handlers/TimeControlHandler.cs:35]: Except setting it in the controls, from what i found

**Comment by garrettluskey** [comment]
Currently, this PR does not fully satisfy #687

Please update to include all functionality

**Comment by EgardA** [comment]
But it does.. I can stay on fast forward no matter what action I take. Let me know if you have found any actions that does pause or slow it down.

I can however agree on that it can be done in a better way, like change the message at the origin instead of in passing.

**Review by garrettluskey — APPROVED** [review]

**Pull request merged**